docs/generator_template.md

```diff
@@ -0,0 +1,568 @@
+
+```yml
+servers:
+  dev:
+    url: test.mosquitto.org
+    protocol: mqtt
+    bindings:
+      mqtt:
+        clientId: temperature-service
+```
+
+This will allow you to also define the broker you will use in production in the servers section above.
+Therefore, we can template the code `mqttBroker = 'test.mosquitto.org'` in **index.js** so the value is populated dynamically at runtime depending on the specified server environment.
+
+Generator has a **parameters** object used to define parameters you use to dynamically to modify your template code at runtime. It also supports the **server** parameter that defines the server configuration value. Navigate to **package.json** and add the snippet below:
+
+```json
+    "generator": {
+        # ...(redacted for brevity)
+        "parameters": {
+            "server": {
+              "description": "The server you want to use in the code.",
+              "required": true
+            }
+        }
+    }
+```
+
+`"required": true`: makes the parameter mandatory and once user forgets to add it to the cli command, a proper error message is yielded.
+You'll pass the server to be used to generate your code using `--param server=dev` in the AsyncAPI CLI command. Failure to which you'll get an error:
+
+```cmd
+Generator Error: This template requires the following missing params: server.
+```
+
+Update your `test:generate` script in `package.json` to inlcude the server param `test:generate": "asyncapi generate fromTemplate test/fixtures/asyncapi.yml ./ --output test/project --force-write --param server=dev "`
+You can now replace the static broker from `mqttBroker = 'test.mosquitto.org'` to `mqttBroker = "${asyncapi.servers().get(params.server).url()}"` in **index.js**.
+
+Now the template code looks like:
+
+``` js
+import { File } from '@asyncapi/generator-react-sdk';
+
+// notice that now the template not only gets the instance of parsed AsyncAPI document but also the parameters
+export default function ({ asyncapi }) {
+  console.log(asyncapi, params)
+  return (
+    <File name="client.py">
+      {`import paho.mqtt.client as mqtt
+
+mqttBroker = "${asyncapi.servers().get(params.server).url()}"
+
+class TemperatureServiceClient:
+    def __init__(self):
+        self.client = mqtt.Client()
+        self.client.connect(mqttBroker)
+
+
+    def sendTemperatureChange(self, id):
+        topic = "temperature/changed"
+        self.client.publish(topic, id)`}
+    </File>
+  )
+}
+```
+
+Run `npm test` to validate that your code still works as expected.
+
+### Templating index.js with React
+
+Python takes indentation very seriously and out generated output will be python code. We therefpre need to make sure the indentation in index.js looks right so the generated code is indented properly. After templating the code in index.js, it will look like the following code snippet:
```

Comment:
```suggestion
Python takes indentation very seriously, and our generated output will be Python code. We, therefore, need to make sure the indentation in index.js looks right so the generated code is indented correctly. After templating the code in index.js, it will look like the following code snippet:
```